### PR TITLE
[RFC] vim-patch:8.0.{647,768,797,1085,1092,1107,1133,1408}

### PR DESCRIPTION
**Pull request by janlazo**
**vim-patch:8.0.0647: syntax highlighting can make cause a freeze**
Problem:    Syntax highlighting can make cause a freeze.
Solution:   Apply 'redrawtime' to syntax highlighting, per window.
vim/vim@06f1ed2

**vim-patch:8.0.0768: terminal window status shows "[Scratch]"**
Problem:    Terminal window status shows "[Scratch]".
Solution:   Show "[Terminal]" when no title was set. (Yasuhiro Matsumoto)
            Store the terminal title that vterm sends and use it.  Update the
            special buffer name.  (closes vim/vim#1869)
vim/vim@2155441

**vim-patch:8.0.0797: finished job in terminal window is not handled**
Problem:    Finished job in terminal window is not handled.
Solution:   Add the scrollback buffer.  Use it to fill the buffer when the job
            has ended.
vim/vim@d85f271

**vim-patch:8.0.1085: terminal debugger can't set breakpoints**
Problem:    The terminal debugger can't set breakpoints.
Solution:   Add :Break and :Delete commands.  Also commands for stepping
            through code.
vim/vim@e09ba7b

**vim-patch:8.0.1092: terminal debugger can't evaluate expressions**
Problem:    Terminal debugger can't evaluate expressions.
Solution:   Add :Evaluate and K.  Various other improvements.
vim/vim@45d5f26

**vim-patch:8.0.1107: terminal debugger jumps to non-existing file**
Problem:    Terminal debugger jumps to non-existing file.
Solution:   Check that the file exists.  Add an option to make the Vim wide
            wide. Fix removing highlight groups.
vim/vim@38baa3e

**vim-patch:8.0.1133: syntax timeout not used correctly**
Problem:    Syntax timeout not used correctly.
Solution:   Do not pass the timeout to syntax_start() but set it explicitly.
            (Yasuhiro Matsumoto, closes vim/vim#2139)
vim/vim@f3d769a

**vim-patch:8.0.1408: crash in setqflist()**
Problem:    Crash in setqflist().
Solution:   Check for string to be NULL. (Dominique Pelle, closes vim/vim#2464)
vim/vim@a0ca7d0

Retry of #8662
~~Will followup with [vim-patch:8.0.1133]~~ Done. (https://github.com/vim/vim/commit/f3d769a585040ac47f7054057758809024ef6377). `+toolbar` is N/A, right?

### CODE REVIEW

**Comment by justinmk** [comment]
> `+toolbar` is N/A, right?

Let's say yes. If plugins end up needing it, can revisit later.

**Review by justinmk — APPROVED** [review]

**Comment by mhinz** [comment]
@janlazo Sorry, but 8.1.1177 already got merged. I created a PR for it around the same it was added to this one and thus I missed it.

**Comment by janlazo** [comment]
```
[----------] Running tests from /home/travis/build/neovim/neovim/test/functional/ex_cmds/digraphs_spec.lua
[ RUN      ] :digraphs displays digraphs: -- Output to stderr:
Uncaught Error: ...m-deps/usr/share/lua/5.1/nvim/msgpack_rpc_stream.lua:84: invalid msgpack-rpc string
stack traceback:
	[C]: in function 'receive'
	...m-deps/usr/share/lua/5.1/nvim/msgpack_rpc_stream.lua:84: in function 'cb'
	...deps/usr/share/lua/5.1/nvim/child_process_stream.lua:43: in function <...deps/usr/share/lua/5.1/nvim/child_process_stream.lua:39>
	[C]: in function 'run'
	.../travis/nvim-deps/usr/share/lua/5.1/nvim/session.lua:179: in function '_run'
	.../travis/nvim-deps/usr/share/lua/5.1/nvim/session.lua:126: in function 'run'
	test/functional/helpers.lua:224: in function 'run_session'
	test/functional/ui/screen.lua:503: in function '_wait'
	test/functional/ui/screen.lua:366: in function 'expect'
	...vim/neovim/test/functional/ex_cmds/digraphs_spec.lua:25: in function <...vim/neovim/test/functional/ex_cmds/digraphs_spec.lua:23>
	...
	.../travis/nvim-deps/usr/share/lua/5.1/busted/block.lua:155: in function 'execute'
	...e/travis/nvim-deps/usr/share/lua/5.1/busted/init.lua:7: in function 'executor'
	...e/travis/nvim-deps/usr/share/lua/5.1/busted/core.lua:312: in function <...e/travis/nvim-deps/usr/share/lua/5.1/busted/core.lua:312>
	[C]: in function 'xpcall'
	...e/travis/nvim-deps/usr/share/lua/5.1/busted/core.lua:178: in function 'safe'
	...e/travis/nvim-deps/usr/share/lua/5.1/busted/core.lua:312: in function 'execute'
	...ravis/nvim-deps/usr/share/lua/5.1/busted/execute.lua:58: in function 'execute'
	...travis/nvim-deps/usr/share/lua/5.1/busted/runner.lua:174: in function <...travis/nvim-deps/usr/share/lua/5.1/busted/runner.lua:11>
	.../usr/lib/luarocks/rocks/busted/2.0.rc12-1/bin/busted:3: in main chunk
	[C]: ?
CMake Error at /home/travis/build/neovim/neovim/cmake/RunTests.cmake:56 (message):
  functional tests failed with error: 1
```

**Comment by justinmk** [comment]
@janlazo `digraphs_spec` failure is known issue https://github.com/neovim/neovim/issues/8102 .

Is this RDY?

File: src/nvim/screen.c [hunk 1]
```diff
@@ -2587,6 +2594,9 @@ win_line (
     }
     next_search_hl(wp, shl, lnum, (colnr_T)v,
                    shl == &search_hl ? NULL : cur);
+    if (wp->w_s->b_syn_slow) {
+      has_syntax = false;
+    }
```
Comment (lewis6991) [inline src/nvim/screen.c:2599]: This looks like this was added by mistake.

**Pull request merged**